org.cryptomator.Cryptomator.yaml

```diff
@@ -0,0 +1,100 @@
+    sources:
+      - type: archive
+        url: https://github.com/libfuse/libfuse/releases/download/fuse-2.9.9/fuse-2.9.9.tar.gz
+        sha256: d0e69d5d608cc22ff4843791ad097f554dd32540ddc9bed7638cc6fea7c1b4b5
+      - type: file
+        path: build-aux/fusermount-wrapper.sh
+  - name: openjdk
+    buildsystem: simple
+    build-commands:
+      - /usr/lib/sdk/openjdk17/install.sh
+  - name: cryptomator
+    buildsystem: simple
+    build-options:
+      env:
+        PATH: /app/bin:/usr/bin:/usr/lib/sdk/openjdk17/bin
+        MAVEN_OPTS: -Dmaven.repo.local=.m2/repository
+        JAVA_HOME: /usr/lib/sdk/openjdk17/jvm/openjdk-17
+        VERSION: 1.6.5
+        REVISION_NO: '1'
+    build-commands:
+      - mvn clean package -DskipTests -Plinux
+      - cp dist/linux/launcher* target
+      - cp target/cryptomator-*.jar target/mods
+      - cd target
+      - jlink
+        --output runtime
+        --module-path "${JAVA_HOME}/jmods"
+        --add-modules java.base,java.desktop,java.logging,java.naming,java.net.http,java.scripting,java.sql,java.xml,jdk.unsupported,jdk.crypto.ec,jdk.accessibility
+        --no-header-files
+        --no-man-pages
+        --strip-debug
+        --compress=1
+      - jpackage
+        --type app-image
+        --runtime-image runtime
+        --input target/libs
+        --module-path target/mods
+        --module org.cryptomator.desktop/org.cryptomator.launcher.Cryptomator
+        --dest .
+        --name Cryptomator
+        --vendor 'Skymatic GmbH'
+        --copyright '(C) 2016 - 2022 Skymatic GmbH'
+        --java-options '-Xss5m'
+        --java-options '-Xmx256m'
+        --java-options '-Dfile.encoding='utf-8''
+        --java-options '-Dcryptomator.logDir='~/.local/share/Cryptomator/logs''
+        --java-options '-Dcryptomator.settingsPath='~/.config/Cryptomator/settings.json:~/.Cryptomator/settings.json''
+        --java-options '-Dcryptomator.ipcSocketPath='~/.config/Cryptomator/ipc.socket''
+        --java-options '-Dcryptomator.mountPointsDir='~/.local/share/Cryptomator/mnt''
+        --java-options '-Dcryptomator.showTrayIcon=false'
+        --java-options "-Dcryptomator.buildNumber='flatpak-${REVISION_NO}'"
+        --java-options "-Dcryptomator.appVersion='${VERSION}'"
+        --app-version "${VERSION}.${REVISION_NO}"
+        --verbose
+      - cp -R Cryptomator /app/
+      - ln -s /app/Cryptomator/bin/Cryptomator /app/bin/cryptomator
+      - install -D -m0644 -t /app/share/applications/ org.cryptomator.Cryptomator.desktop
+      - install -D -m0644 -t /app/share/icons/hicolor/scalable/apps/ org.cryptomator.Cryptomator.svg
+      - install -D -m0644 -t /app/share/metainfo/ org.cryptomator.Cryptomator.appdata.xml
+    sources:
+      - maven-dependencies.yaml
+      - type: file
+        path: org.cryptomator.Cryptomator.appdata.xml
+      - type: file
+        path: org.cryptomator.Cryptomator.desktop
+      - type: file
+        path: org.cryptomator.Cryptomator.svg
```

Comment:
you should use the file from upstream instead of having them here.

Reply:
svg is the same, appdata diffs can be merged upstream, however the desktop file differs. I see two options:

1. have a flatpak-specific .desktop file in this repo
2. use upstream .desktop file and apply a patch during each build

Reply:
differ how? patching the desktop file is common, but it can also be avoided.

Reply:
```diff
[Desktop Entry]
Categories=Utility;FileTools;Security;
Comment=Cloud storage encryption utility
-Exec=cryptomator %F
+Exec=cryptomator %U
Icon=org.cryptomator.Cryptomator
Keywords=encryption;
-MimeType=application/vnd.cryptomator.encrypted;application/vnd.cryptomator.vault;
Name=Cryptomator
-StartupWMClass=org.cryptomator.launcher.Cryptomator
+StartupWMClass=skymatic-cryptomator
+StartupNotify=true
Terminal=false
Type=Application
-Version=1.6.5
```

I guess the version is not desired in Flatpaks as it is part of the package metadata? And no idea why the different `StartupWMClass`. Arguments passed to `Exec` as URLs rather than paths may be due to the sandbox environment?

I honestly don't know _why_ there is a different desktop file, so if you can't explain this either, I'd suggest to use the upstream versions for now and release patches should the need arise.

Reply:
For starter `Version` is not what you think it is:

https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s06.html :
>  Version of the Desktop Entry Specification that the desktop entry conforms with. Entries that confirm with this version of the specification should use 1.5. Note that the version field is not required to be present. 

`MimeType` should be in there.
No reason the `Exec` line be different in term of paramaters.

So it's safe to say upstream should be fine.